pokemon-league-participation-manager/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/entities/Gym.java

```diff
@@ -0,0 +1,101 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.entities;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.enums.PokemonType;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+
+/**
+ * Entity class for object gym.
+ *
+ * @author Michal Mokros 456442
+ */
+@Entity(name = "Gym")
+@Table(name = "GYM")
+public class Gym {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    private String location;
+
+    @NotNull
+    @Enumerated
+    private PokemonType type;
+
+    @NotNull
+    @OneToOne
+    @JoinColumn(name = "trainerId")
+    private Trainer gymLeader;
+
```

Comment:
Might as well add the @ OneToMany Set of Badges/Challenges, don't forget to tell the JPA about the @ ManyToOne direction - see https://vladmihalcea.com/the-best-way-to-map-a-onetomany-association-with-jpa-and-hibernate/ "Bidirectional @ OneToMany".

Reply:
On the diagrams document it says that only one trainer can be the leader of given gym at one time. So I don't really understand the change here.

Reply:
It's the discussion we had on facebook with @jirmed about including the @ OneToMany directions of the entity relationships. Since we have a use case for revoking badges by gym leader, they will need to see a list of badges issued by their gym, hence the inclusion of a lazy loaded set of the badges issued by the gym.

Reply:
So I understand it well that there should be another variable - set of Badges with @OneToMany annotation right?

Reply:
Yes. that is indeed correct.

Reply:
The association (on badges) is bidirectional, mappedBy is then a mandatory argument (https://docs.oracle.com/javaee/7/api/javax/persistence/OneToMany.html). Also the cascading needs to be set (no cascading is incorrect - at least persist should cascade).